### PR TITLE
Feature/4 manage books

### PR DESCRIPTION
**Pull request by lukavi44**

### CODE REVIEW

File: library-app/src/components/Books/BooksItem/BooksItem.tsx [hunk 1]
```diff
@@ -1,13 +1,64 @@
...
+
+  useEffect(() => {
+    setCoverPlaceholder(imgPlaceholder)
+  }, [])
+
+  // const editBookHandler = (event: MouseEvent) => {
+  //   event.preventDefault()
+  //   setIsEditBook(!isEditBook)
+  // }
+
```
Comment (aleksandarkrnjaic) [inline library-app/src/components/Books/BooksItem/BooksItem.tsx:22]: commented code

File: library-app/src/components/Layout/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const portalDiv = document.getElementById('portal') as HTMLElement
+
+export interface ModalProps {
+  // open: boolean
+  children: React.ReactNode
+  onClose: React.MouseEventHandler
+}
+
+const Modal = ({ children, onClose }: ModalProps) => {
+  // if (!open) return null
```
Comment (aleksandarkrnjaic) [inline library-app/src/components/Layout/Modal.tsx:13]: commented code

File: library-app/src/components/Layout/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import ReactDOM from 'react-dom'
+import styles from './Modal.module.css'
+
+const portalDiv = document.getElementById('portal') as HTMLElement
+
+export interface ModalProps {
+  // open: boolean
```
Comment (aleksandarkrnjaic) [inline library-app/src/components/Layout/Modal.tsx:7]: commented code

File: library-app/src/helpers/manageLocalStorage.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+const accessTokenKey = 'accessToken'
+const refreshTokenKey = 'refreshToken'
+const expirationKey = 'expiration'
+
+export const setLocalStorage = ({ AccessToken, RefreshToken, Expiration }: LocalStorage) => {
+  localStorage.setItem(accessTokenKey, AccessToken)
+  localStorage.setItem(refreshTokenKey, RefreshToken)
+  localStorage.setItem(expirationKey, Expiration)
```
Comment (aleksandarkrnjaic) [inline library-app/src/helpers/manageLocalStorage.tsx:14]: Alternatively, you can hold whole object in localstorage.setItem(JSON.stringify(object))

File: library-app/src/helpers/manageLocalStorage.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+export interface LocalStorage {
+  AccessToken: string
+  RefreshToken: string
+  Expiration: string
+}
```
Comment (aleksandarkrnjaic) [inline library-app/src/helpers/manageLocalStorage.tsx:5]: LocalStorage named type implies this is a type for localStorage object

File: library-app/src/helpers/manageLocalStorage.tsx (hunk 4 above)
Comment (aleksandarkrnjaic) [reply library-app/src/helpers/manageLocalStorage.tsx:14]: getting would go like this: JSON.parse(localStorage.getItem('objectNameInStorage')).

File: library-app/src/App.tsx [hunk 6]
```diff
@@ -2,11 +2,24 @@ import Homepage from './components/pages/Homepage/Homepage'
 import styles from './App.module.css'
 import { Route, Routes } from 'react-router-dom'
 import Login from './components/pages/Login/Login'
+import React, { useEffect, useState } from 'react'
+import PrivateRoutes from './router/PrivateRoutes'
+import { getAccessToken } from './helpers/manageLocalStorage'
+import BookDetails from './components/pages/BookDetails/BooksDetails'
 
 function App() {
+  const [accessToken, setAccessToken] = useState<string | null>('')
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/App.tsx:11]: What is null used for if the initial state is empty string?

File: library-app/src/components/Books/BooksList/ManageBookForm.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    getAuthors().then((response) => {
+      setAuthors(response.data)
+    })
+  }, [])
+
+  useEffect(() => {
+    try {
+      fetchAuthorsData()
+    } catch (error) {
+      if (error) console.log('nema autora')
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/components/Books/BooksList/ManageBookForm.tsx:51]: What is this 'if' used for?

File: library-app/src/components/Books/BooksList/ManageBookForm.tsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+      form.append('Quantity', formData.Quantity.toString())
+      form.append('Title', formData.Title)
+      formData.AuthorIds.forEach((author) => form.append('AuthorIds', author.Id.toString()))
+
+      await postBookRequest(form)
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response?.status === 401) {
+        console.log('neautorizovan')
+      }
+      return
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/components/Books/BooksList/ManageBookForm.tsx:72]: Why return?

File: library-app/src/components/Sidebar/Sidebar.tsx [hunk 9]
```diff
@@ -19,38 +23,42 @@ const Sidebar = ({ isLoggedIn }: MainLayoutProps) => {
...
-              </a>
-            </div>
-            <div className={styles['btn-holder']}>
-              <a>
-                <img src={account} alt='admin option2' />
-              </a>
+              <button
+                className={styles['add-new-book']}
+                type='submit'
+                onClick={() => setIsModalOpened(true)}
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/components/Sidebar/Sidebar.tsx:50]: Type submit?

File: library-app/src/components/pages/Login/Login.tsx [hunk 10]
```diff
@@ -22,11 +23,12 @@ const Login = () => {
   }, [])
 
   const handleEmailChange = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
-    if (currentTarget.value.trim() === '') {
+    if (currentTarget.value === '') {
       setEnteredEmailIsValid(false)
+    } else {
+      setEnteredEmailIsValid(true)
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/components/pages/Login/Login.tsx:29]: Think about the way to write this if-else in one line

File: library-app/src/components/pages/Login/Login.tsx [hunk 11]
```diff
@@ -57,13 +59,11 @@ const Login = () => {
...
+      return
     } else {
       await LoginRequest(formData)
         .then(({ data }) => {
-          localStorage.setItem('accessToken', data.accessToken)
-          localStorage.setItem('refreshToken', data.refreshToken)
-          localStorage.setItem('expiration', data.expiration)
+          setLocalStorage(data)
           navigateTo('/')
           console.log(data)
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/components/pages/Login/Login.tsx:68]: console.log

File: library-app/src/services/BooksServices.tsx [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import { AxiosResponse } from 'axios'
+import { BookBodyData, BookBodyDataGet } from '../models/bookData.model'
+import axiosInstance from './axiosConfig'
+
+export const postBookRequest = (body: FormData): Promise<AxiosResponse> => {
+  return axiosInstance.post<BookBodyData>('api/Books', body, {
+    headers: {
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/services/BooksServices.tsx:7]: You don't need this if you have interceptors

File: library-app/src/services/BooksServices.tsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+      Authorization: `Bearer ${localStorage.getItem('accessToken')}`,
+    },
+  })
+}
+
+export const getBooksRequest = (): Promise<AxiosResponse> => {
+  return axiosInstance.get<BookBodyDataGet>('api/Books', {
+    headers: {
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/services/BooksServices.tsx:15]: Same

**Pull request closed without merging**